File: 프로그래머스/3/43105. 정수 삼각형/정수 삼각형.java
import java.util.*;

class Solution {
    public int solution(int[][] triangle) {

        int arr_l = triangle.length;

        for (int i = 1; i < arr_l; i++) {
            for (int j = 0; j < triangle[i].length; j++) {
                if (j == 0) { //왼쪽 끝 인덱스
                    triangle[i][j] += triangle[i - 1][j];
                } else if (j == triangle[i].length - 1) { //오른쪽 끝 인덱스
                    triangle[i][j] += triangle[i - 1][j - 1];
                } else {
                    triangle[i][j] = Math.max(triangle[i][j] + triangle[i - 1][j], triangle[i][j] + triangle[i - 1][j - 1]);
                }
            }
        }

        return Arrays.stream(triangle[arr_l-1]).max().getAsInt();
    }
}

File: 프로그래머스/3/43105. 정수 삼각형/정수 삼각형.py
def solution(triangle):
    
    arr_l = len(triangle)
    
    for i in range(1, arr_l):
        for j in range(len(triangle[i])):
            if(j == 0): #가장 왼쪽 인덱스
                triangle[i][j] += triangle[i-1][j]
            elif(j == len(triangle[i]) - 1): #가장 오른쪽 인덱스
                triangle[i][j] += triangle[i-1][j-1]
            else:
                triangle[i][j] = max(triangle[i][j] + triangle[i-1][j-1], triangle[i][j] + triangle[i-1][j])
    return max(triangle[arr_l-1])